### PR TITLE
Email extension for Quarto

### LINKED ISSUE #4946: Email Integration
Work for this is being done as a Quarto extension in the `https://github.com/rich-iannone/quarto-email` repository (private for now). The email body will be in a div with class `"email"`.

To make this work with the Connect emailing system, the main goal, the filter needs to generate a JSON file in the working directory with specific key names. Connect will be updated to watch for this file; this is close to the blastula/Connect model where R Markdown metadata is captured when certain key-value pairs are present. 
```[tasklist]
### Tasks
- [x] add `attachments` as resources
- [x] test email messages with various image types
- [x] test email messages with external (web) images
```

### PR DESCRIPTION
**Pull request by rich-iannone**
This contains the extension `email`, which is a filter that can use an `.email`-classed div as content for an HTML email through Posit Connect (via `rsconnect` for R/Python).

Fixes: https://github.com/quarto-dev/quarto-cli/issues/4946

### CODE REVIEW

File: r_report_email_format_files/figure-html/diamonds_plot-1.png
Comment (dragonstyle) [inline r_report_email_format_files/figure-html/diamonds_plot-1.png:1]: I'm guessing that all the files in `r_report_email_format_files` are inadvertently included and should be excluded from this PR (can you confirm?).

File: src/resources/filters/quarto-post/email.lua [hunk 1]
```diff
@@ -0,0 +1,423 @@
...
+    connect_report_subscription_url
+  )
+
+  -- For each of the <img> tags we need to create a Base64-encoded representation
+  -- of the image and place that into the table `email_images` (keyed by `cid`)
+
+  local image_data = nil
+
+  for cid, img in pairs(image_tbl) do
+    if (true) then
```
Comment (dragonstyle) [inline src/resources/filters/quarto-post/email.lua:334]: Guessing this might just be cruft...

**Review by dragonstyle — COMMENTED** [review]
This looks great-  I made two very minor comments that I think you should take a quick look at (more hygiene than real changes).

File: r_report_email_format_files/figure-html/diamonds_plot-1.png
Comment (rich-iannone) [reply r_report_email_format_files/figure-html/diamonds_plot-1.png:1]: Yes, committed that by mistake. Now removed!

File: src/resources/filters/quarto-post/email.lua (hunk 1 above)
Comment (rich-iannone) [reply src/resources/filters/quarto-post/email.lua]: I removed the `if (true) then ...` conditional. Thanks!

**Comment by rich-iannone** [comment]
Seems like there's another issue that cropped up. In the integrated extension, the div with the `.email` class isn't omitted from the main report file. Also, the preview .html file doesn't get written. I'll take a look into this.

**Comment by cscheid** [comment]
This is huge! 🎉

**Comment by cderv** [comment]
Awesome work @rich-iannone ! Thank you !

**Pull request merged**